File: documentation-samples/quickstarts/change-model/php/add-utterances.php
<?php

// NOTE: Be sure to uncomment the following line in your php.ini file.
// ;extension=php_openssl.dll

// **********************************************
// *** Update or verify the following values. ***
// **********************************************

// NOTE: Replace this example LUIS application ID with the ID of your LUIS application.
$appID = "YOUR-APP-ID";

// NOTE: Replace this example LUIS application version number with the version number of your LUIS application.
$appVersion = "0.1";

// NOTE: Replace this example LUIS authoring key with a valid key.
$key = "YOUR-AUTHORING-KEY";

$host = "https://westus.api.cognitive.microsoft.com";
$path = "/luis/api/v2.0/apps/" . $appID . "/versions/" . $appVersion . "/";
$uri = $host . $path;
$file = "utterances.json";

function SendGet ($uri, $key) {

	echo "GET " . $uri;

	$headers = "Content-type: text/json\r\n" .
		"Ocp-Apim-Subscription-Key: $key\r\n";

	// NOTE: Use the key 'http' even if you are making an HTTPS request. See:
	// http://php.net/manual/en/function.stream-context-create.php
	$options = array (
		'http' => array (
			'header' => $headers,
			'method' => 'GET',
		)
	);
	// Perform the Web request and get the JSON response
	$context  = stream_context_create ($options);
	$result = file_get_contents ($uri, false, $context);
	return $result;
}

function SendPost ($uri, $key, $content) {

	echo "POST " . $uri;

	$headers = "Content-type: text/json\r\n" .
		"Ocp-Apim-Subscription-Key: $key\r\n";

	// NOTE: Use the key 'http' even if you are making an HTTPS request. See:
	// http://php.net/manual/en/function.stream-context-create.php
	$options = array (
		'http' => array (
			'header' => $headers,
			'method' => 'POST',
			'content' => $content
		)
	);
	// Perform the Web request and get the JSON response
	$context  = stream_context_create ($options);
	$result = file_get_contents ($uri, false, $context);
	return $result;
}

function AddUtterances($uri, $key, $input_file) {
	$content = file_get_contents ($input_file);
	echo "Added utterances.\n";
	$result = SendPost ($uri . "examples", $key, $content);
    echo json_encode (json_decode ($result), JSON_PRETTY_PRINT);
}

function Train($uri, $key) {
	echo "Sent training request.\n";
	$content = null;
	$result = SendPost ($uri . "train", $key, $content);
    echo json_encode (json_decode ($result), JSON_PRETTY_PRINT);
	echo "\n";
	Status($uri, $key);
}

function Status($uri, $key) {
	echo "Requested training status.\n";
	$result = SendGet ($uri . "train", $key);
    echo json_encode (json_decode ($result), JSON_PRETTY_PRINT);
}

AddUtterances($uri, $key, $file);
Train($uri, $key);
Status($uri, $key);

?>

File: documentation-samples/quickstarts/change-model/go/add-utterances.go

// dependencies
package main
import (
	"fmt"
	"net/http"
	"io/ioutil"
	"log"
	"strings"
)

// generic HTTP request
// includes setting header with authoring key
func httpRequest(httpVerb string, url string, authoringKey string, body string){

		client := &http.Client{}
	
		request, err := http.NewRequest(httpVerb, url, strings.NewReader(body))
		request.Header.Add("Ocp-Apim-Subscription-Key", authoringKey)

		fmt.Println("body")
		fmt.Println(body)

		response, err := client.Do(request)
		if err != nil {
			log.Fatal(err)
		} else {
			defer response.Body.Close()
			contents, err := ioutil.ReadAll(response.Body)
			if err != nil {
				log.Fatal(err)
			}
			fmt.Println("   ", response.StatusCode)
			fmt.Println(string(contents))
		}
}

// main function
func main() {

	// NOTE: change to your app ID
	var appID = "YOUR-APP-ID"

	// NOTE: change to your authoring key
	var authoringKey = "YOUR-AUTHORING-KEY"

	var version = "0.1"

	var exampleUtterances = "utterances.json"

	fmt.Println("add example utterances requested")
	addUtterance(authoringKey, appID, version, exampleUtterances)

	fmt.Println("training selected")
	requestTraining(authoringKey, appID,  version)

	fmt.Println("training status selected")
	getTrainingStatus(authoringKey, appID, version)

}

// get utterances from file and add to model
func addUtterance(authoringKey string, appID string,  version string, fileOfLabeledExampleUtterances string){

    exampleUtterancesAsBytes, err := ioutil.ReadFile(fileOfLabeledExampleUtterances) // just pass the file name
    if err != nil {
        fmt.Print(err)
	}
	fmt.Println(string(exampleUtterancesAsBytes))


	// NOTE: region is westus
	var authoringUrl = fmt.Sprintf("https://westus.api.cognitive.microsoft.com/luis/api/v2.0/apps/%s/versions/%s/examples", appID, version)

	httpRequest("POST", authoringUrl, authoringKey, (string(exampleUtterancesAsBytes)))
}
func requestTraining(authoringKey string, appID string,  version string){

	trainApp("POST", authoringKey, appID, version)
}
func trainApp(httpVerb string, authoringKey string, appID string,  version string){

	var authoringUrl = fmt.Sprintf("https://westus.api.cognitive.microsoft.com/luis/api/v2.0/apps/%s/versions/%s/train", appID, version)

	httpRequest(httpVerb,authoringUrl, authoringKey, "")
}
func getTrainingStatus(authoringKey string, appID string, version string){

	trainApp("GET", authoringKey, appID, version)
}